feat: transition cetgenvvs to module (#503)
vmware/power-validated-solutions-for-cloud-foundation: docs/documentation/functions/certificates/Invoke-GeneratePrivateKeyAndCsr.md

# Invoke-GeneratePrivateKeyAndCsr

## Synopsis

> Generate Private Key and certificate signing request (CSR) files using Openssl with a common name and neccesary parameters.

Reviewer: ```suggestion
Generate the private key and certificate signing request (CSR) files using OpenSSL with a common name and additional parameters as necessary.
```

Author: addressed in commit cfcd77a79eb20278e44b123b0092c173dd7b5014

## Syntax

```powershell
Invoke-GeneratePrivateKeyAndCsr [-certDirPath] <String> [-subjectAlternativeNames] <String> [-keySize] <Int>
 [-expireDays] <Int> [-organization] <String> [-organizationUnit] <String> [-locality] <String>
 [-state] <String> [-country] <String>
```

## Description

> The `Invoke-GeneratePrivateKeyAndCsr` cmdlet generates the private key and certificate signing request (CSR) files using Openssl with a common name and neccesary parameters.

Reviewer: ```suggestion
The `Invoke-GeneratePrivateKeyAndCsr` cmdlet generates the private key and certificate signing request (CSR) files using OpenSSL with a common name and neccesary parameters.
```

Author: addressed in commit cfcd77a79eb20278e44b123b0092c173dd7b5014

## Examples

### Example 1

```powershell
Invoke-GeneratePrivateKeyAndCsr -outDirPath "C:\certificates" -commonName "sfo-vrli01.rainpole.io" -subjectAlternativeNames "sfo-vrli01a.rainpole.io, sfo-vrli01b.rainpole.io, sfo-vrli01c.rainpole.io" -keySize 4096 -expireDays 720 -organization "rainpole" -organizationUnit "IT" -locality "San Francisco" -state "California" -country "US"
```

> This example will generate a private key and CSR pair for VMware Aria Operations for Logs where the integrated load balancer has a common name of sfo-vrli01.rainpole.io and includes SANs for each cluster node's fully qualified domain name value. 

Reviewer: Should these examples be sfo.rainmpole.io?

Author: addressed in commit c96a0aa15eab993f1b7f284e5e402712b1a9401f

## Parameters

### -outDirPath

> The directory path to store the signed certificate, private key and certificate signing request file.

Reviewer: ```suggestion
The directory path to store the signed certificate, private key, and certificate signing request file.
```

Author: addressed in commit c96a0aa15eab993f1b7f284e5e402712b1a9401f

```yaml
Type: String
Parameter Sets: (All)
Aliases:

Required: True
Position: 1
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -commonName

The common name for the signed certificate request.

```yaml
Type: String
Parameter Sets: (All)
Aliases:

Required: True
Position: 2
Default value: None
Accept pipeline input: False
Accept wildcard characters: False
```

### -subjectAlternativeNames

> The subject alternative names (SAN) for the signed certificate request.  Separate each SAN with comma.

Reviewer: ```suggestion
The subject alternative names (SAN) for the signed certificate request. Separate each SAN with comma.
```

Author: addressed in commit c96a0aa15eab993f1b7f284e5e402712b1a9401f
